prov/rxm/src/rxm_cq.c

```diff
@@ -501,6 +501,41 @@ ssize_t rxm_rndv_read(struct rxm_rx_buf *rx_buf)
 	return ret;
 }
 
+static ssize_t rxm_rndv_write(struct rxm_rx_buf *rx_buf)
+{
+	int i;
+	ssize_t ret;
+	struct rxm_tx_rndv_buf *tx_buf = ofi_bufpool_get_ibuf(
+		rx_buf->ep->buf_pools[RXM_BUF_POOL_TX_RNDV].pool,
+		rx_buf->pkt.ctrl_hdr.msg_id);
+	size_t total_len = tx_buf->pkt.hdr.size, rma_len = 0;
+	struct rxm_rndv_hdr *rx_hdr = (struct rxm_rndv_hdr *) rx_buf->pkt.data;
+
+	tx_buf->write_rndv.remote_hdr.count = rx_hdr->count;
+	for (i = 0; i < tx_buf->write_rndv.remote_hdr.count; i++) {
+		tx_buf->write_rndv.remote_hdr.iov[i].addr = rx_hdr->iov[i].addr;
+		tx_buf->write_rndv.remote_hdr.iov[i].len = rx_hdr->iov[i].len;
+		tx_buf->write_rndv.remote_hdr.iov[i].key = rx_hdr->iov[i].key;
+		if (total_len > rma_len) {
+			tx_buf->write_rndv.rndv_rma_count++;
+			rma_len += tx_buf->write_rndv.remote_hdr.iov[i].len;
+		}
```

Comment:
I'm not understanding the purpose of this if statement.  The total rm'a len should be the same as the total len.  That's going to be limited by max msg size.  The number of rmas being issued should be the same as iov count (unless the provider supports more than 1 iov per rma).

Reply:
This was my original assumption as well, though in testing I found scenarios where the receiver iovs were of different sizes than the senders (the sender's ideally matching the data).  There were cases where the receiver's first iov could completely accept the entirety of the message, but later iovs could not accept much at all.  Below is an example of such a scenario, this is not identical to the numbers I was seeing in testing, but shows the issue i'm trying to address here.

```
Imagine a data set spanning 4 iovs, a total length of 199608.
On the sender side, the data may likely be equally distributed.
On the receiver side, the data may all end up in the first iov 
and therefore would only need a single rma.

sender iov:
[0].len = 49902
[1].len = 49902
[2].len = 49902
[3].len = 49902

reciever iov:
[0].len = 1048576
[1].len = 32768
[2].len = 12500
[3].len = 49
```
 `ofi_copy_iov_desc` handles packing the data into the first iov